src/CometWrapper.sol

```diff
@@ -92,8 +92,12 @@ contract CometWrapper is ERC4626, CometHelpers {
         if (assets == 0) revert ZeroAssets();
         if (msg.sender != owner) {
             uint256 allowed = allowance[owner][msg.sender];
-
-            if (allowed != type(uint256).max) allowance[owner][msg.sender] = allowed - shares;
+            shares = convertToShares(assets);
+            if (allowed < shares) revert InsufficientAllowance();
+            // TODO: Move this to after shares is recomputed
```

Comment:
Should we also remove this comment? 
It looks like it has already moved after share is computed? `convertToShares()` ?

Reply:
This comment is referring to the re-computation of `shares` on line 106. In the current implementation, the `shares` here can be slightly different from the `shares` below, so the allowance will be decremented by a slightly different value.

This TODO is addressed in PR #4.